### DISCUSSION TITLE
The new Taiko HD is dangerous for the game

### DISCUSSION THREAD

**Discussion by chayleaf** [ppy/osu#27325] [Ideas]
Until pp was enabled, I didn't really want to bring this up. I enjoy the challenge of playing Lazer's HD on low bpm, and I thought the community will eventually complain about this enough for something to be done about it. Despite me liking to play it, I think it's dangerous for the game as a whole - at the very least I'm asking for it to be unranked in Lazer for the time being, and in this post I will explain the reason.

First, let me break down the current mods and pp situation - not including star rating, which doesn't matter much for mod balance.

Terms:

- NM - no mod
- HD, DT, HR, EZ - hidden, double time, hard rock, easy mods
- SV - scroll speed
- FC, to FC - full combo, to get a full combo
- acc, to acc - accuracy, to play with high accuracy
- Sightread - first ever try on a map

There's NM (I include DT only in this category), which is somewhat viable for pp, but has its limits. The current number 2 taiko player is a NM player. Now, there's something important to know about NM - the amount of notes visible depends on the player's screen resolution. Since less notes on screen can be easier to read (this works like AR in std), many players play on 4:3 on low bpm to get higher accuracy and consistency, and use 16:9 when they need to be able to read higher SV. Since final SV is primarily affected by BPM, they usually use 16:9 on high BPM and 4:3 on low BPM.

There's HR, which increases SV by 1.4 on 4:3, or by 1.8666 on 16:9. This is actually slightly unbalanced, so people almost always use wider resolutions for HR, even if they play 4:3 NM. HR is often used by low ranked players because HR is fairly easy to learn for up to 200 BPM, but it becomes an uphill battle from there. The highest level players are able to play 300 BPM with HR, but I don't know to what extent memorization is required for them - it certainly doesn't seem sightreadable. Just to give you the idea of how hard it is - a few years ago, [nameless_ll](https://osu.ppy.sh/users/11199742/taiko) got accused of cheating by some people for FCing [silly 300bpm thing](https://osu.ppy.sh/beatmapsets/651668#taiko/1381623) with HR in less than 30 minutes after playing the map for the first time (the first HR play was a B, and eventually it became an S).

In terms of pp, HR doesn't actually affect it, however it increases OD, potentially allowing you to get more accuracy pp, and the increased SV makes maps easier to acc (provided you can read it), this is the same effect switching to 4:3 provides, but on a greater scale.

There's HD, which hides the left part of the playfield, making only the right part of the playfield visible (leaving a constant "visible area"), and forces the playfield to be 4:3 or lower. On low SV (read: BPM), HD is challenging because the time between notes appearing and you having to play them (let's call this "delta") is very high. On high SV (read: BPM), HD doesn't actually hide the notes for a long time, so it becomes trivial if you can read the SV in 4:3. But reading the SV in 4:3 becomes precisely the problem, as it can be noticeably harder than 16:9 - it's roughly equivalent to a 1.4x SV increase (the exact value was there in some of my commits to lazer before being patched away), which is 1.3333x slower than HR, but still can be very challenging.

The difficulty of HD comes from different sources with different SV. Despite that, the pp given doesn't depend on SV at all, it simply gives bonus strain (?) pp. This is somewhat balanced because HD is difficult on both low and high SV, albeit for different reasons, even if HD is pretty easy on mid-range SV.

However, when paired with EZ or HR, the HD mod changes.

With EZHD, SV is reduced, but the delta doesn't change (so the visible area increases). This more or less makes it equivalent to 16:9 in terms of SV reading, but significantly reduces the potential accuracy pp you can get. Still, it can be more challenging than NM because of low SV sections, and because of having to be able to play HD on at least a basic level. At high BPM, EZ also reduces the rhythmic precision required to hit notes, making very dense patterns easier to hit - the same way EZ increases circle size in std. This made EZHD "the" pp combination for high level players, but it is somewhat frowned upon because it's used by players who can't play high bpm on 4:3, or by players who don't have the required accuracy to FC high BPM maps with DT and no EZ.

With HDHR, SV is increased, but the delta, again, doesn't change. This actually pushes the visible note area outside of the 4:3 boundary, which is why the 4:3 limitation is removed and the playfield becomes unlocked again. This is very similar to HR, but with the added difficulty of low note visibility. Because HR is easier on a higher aspect ratio, and because the visible area increases ever so slightly with higher aspect ratios, HDHR players tend to use aspect ratios like 21:9, or even higher. The current number 2 player can play simple maps with HDHR at 300BPM, and this is considered super impressive. I don't know his sightread limit, but it's probably ~270BPM, judging by his Hug and Kill +HDRD play.

For a long time, ranked HDHR was considered stupidly hard because of the very low visibility (even lower on standard aspect ratios), and players only attempted it very rarely. It only became "popular" in the last ~3 years, even now not many players can play it.

But the same can't be said about tournaments. On tournaments, Score V2 is used. With Score V2, the HD delta isn't constant with different mods. Instead of preserving the delta, the visible area is preserved. This truly makes HDHR the combination of HD and HR. As an HD player, when I play freemod tournament maps, I pick either NM, HD, or HDHR. HR isn't an option simply because if I can read the SV with HR, HD becomes "free". This shows how important SV is when considering mod balance.

Now, we switch to Lazer.

For NM, Lazer fixes the playfield size to the equivalent of 16:9. No complaints here, the old mechanics basically allowed the player to adjust SV, which can be considered cheesy. That said, I do want to note that covering the right part of the playfield with a towel or a paper or anything else achieves the same effect as switching to a lower aspect ratio in stable.

For HD, Lazer keeps stable's visible area size, and HDHR/EZHD work the same as in Score V2, which is fine with me - I actually like it, as it can stimulate more players to learn HDHR, and makes EZHD much less viable, which is fair too. However, there's one crucial difference - instead of 4:3, HD becomes 16:9 - see [this post](https://github.com/ppy/osu/discussions/17265) for comparison.

There are multiple problems with this:

- HD is already fairly hard to learn for new players in stable, because HD is hard to play on low BPM. This makes it even harder - instead of making ~170 BPM HD hard, now even ~220 BPM HD can be hard for newcomers.
- In tournaments, there are typically two HD maps - a high SV-focused one and a low SV-focused one, focusing on the two difficulties of HD I mentioned above. HD being 16:9 completely removes the "high SV difficulty" aspect of HD, making it equivalent to NM in that regard. This means tournaments have to adapt accordingly, affecting the tournament pools negatively (much more effort is required to find 2 maps focusing on different aspects of HD that are both low SV).
- Most importantly, this *completely* removes the challenge of HD on high SV, and by extension high BPM. This is *much* worse than EZHD, because full accuracy pp is given, full score is given, etc. This *will* completely break the rankings as soon as more high level players switch to Lazer. It's a ticking bomb. And a pp rebalance isn't the solution, because what's the point of HD if it *essentially* doesn't affect gameplay?

So, to summarize, while I'm fine with the new HDHR even if it's easier than on stable, the other changes make HD unnecessarily hard for low BPM, and unnecessarily easy for high BPM, completely breaking the game balance. This should be *very very very very* heavily scrutinized, instead of the current approach where I introduced HD with stable parity, then someone reverted stable parity because it's "more consistent" or whatever, then I thought "oh well I'm too tired to argue about this, and it doesn't give pp anyway", and now it does give pp which is completely broken.

**Comment by bdach** [comment]
Before panicking too much please check https://github.com/ppy/osu/pull/27136 which attempts to bring back ~~all of the broken stable crap~~ beloved stable mechanics, at least with classic mod active.  Maybe we'll extend that to nomod too I dunno.

**Reply by bdach** [reply]
Also

> I introduced HD with stable parity, then someone reverted stable parity because it's "more consistent" or whatever

Can you *specifically* point out to where this happened? I wanna say we generally don't do this without a good reason?

**Reply by chayleaf** [reply]
https://github.com/ppy/osu/commit/455666ed94af154106caad62e3a5a5fe0316ace9 / #14106

My original PR also scaled the playfield up, but I reverted those changes during the review, though I personally prefer it

also, I'd like to say again that I like the new HD mod - I like it more than the stable HD in some aspects, in fact I've basically played with it for this entire year, I just think it's terribly unbalanced - same as EZHD in stable, but worse
